docs/assembly.rst

```diff
@@ -358,3 +355,16 @@ in memory is automatically considered memory-safe and does not need to be annota
     It is your responsibility to make sure that the assembly actually satisfies the memory model. If you annotate
     an assembly block as memory-safe, but violate one of the memory assumptions, this **will** lead to incorrect and
     undefined behaviour that cannot easily be discovered by testing.
+
+In case you are developing a library that is meant to be compatible across multiple versions
+of solidity, you can use a special comment to annotate an assembly block as memory-safe:
+
+.. code-block:: solidity
+
+    /// @solidity memory-safe-assembly
+    assembly {
+        ...
+    }
+
+Note that this will only be possible until the next breaking release, so if you are not concerned with
```

Comment:
I would propose to keep it at least for 0.9.x

Reply:
Reformulated it to a more neutral "in a future breaking release".